core/tests/puppeteer-acceptance-tests/utility-functions/puppeteer_utils.js

```diff
@@ -0,0 +1,69 @@
+const puppeteer = require("puppeteer");
+const testConstants = require("./testConstants.js");
+module.exports = class browser {
+  page;
+  browser;
+
+  async initialize(){
+    /* currently, headless is set to false and the page viewport
+       is maximized so that it would be easy for the developers
+       to debug easily while testing.
+       We can remove these settings before merging as we have
+       to run the tests in headless mode. */
+    await puppeteer
+      .launch({
+        headless: false,
+        args: ["--start-fullscreen", "--use-fake-ui-for-media-stream"]
+      })
+      .then(async (browser) => {
+        this.browser = await browser;
+        this.page = await browser.newPage();
+        await (this.page).setViewport({ width: 0, height: 0 });
+      });
+
+      return this.page;
+  }
+
+  async signInWithEmail(email) {
+    await this.goto(testConstants.URLs.home);
+    await this.clickOn("button", "OK");
+    await this.clickOn("span", "Sign in");
+    await this.type(testConstants.SignInDetails.inputField, email);
+    await this.clickOn("span", "Sign In");
+  }
+  
+  async clickOn(tag, selector, time = 0) {
```

Comment:
Please add full JSDoc to each of these functions so that other devs will know how to use them. Ditto for any function in a "utility" file.